### ISSUE TITLE
fight-(attack) move borked

### ISSUE THREAD

**Issue by mojjj** [ZeroK-RTS/Zero-K#1966]
if you use them with F-move (now attack-, previously fight-move)

subject: [vandal, placeholder]
units ignore static defences and walk into it to be killed to death.

subject: [rogue]
enemy units, caught in placeholder forcefield, are targeted badly. rogue shots miss most of the time.

see http://imgur.com/a/gslGe
from http://zero-k.info/Battles/Detail/447496

**Comment by GoogleFrog** [comment]
What is "them"? Which units are the ones being given the order? What are subjects? Say how to reproduce this.

**Comment by mojjj** [comment]
user placeholder, vandal and send them with f-move towards enemy static defence

user rogue and send them with f-move towards enemy units caught in placeholder forcefield

**Comment by Licho1** [comment]
you pressed F ? F is force attack.

**Comment by mtroyka** [comment]
Yes rogues are terrible at hitting things caught by placeholder. Rockos and scalpels (and oddly, hammers) are much better at it, and I don't think that anything can be done about it in general.

Vandals are not the only AA unit that doesn't avoid enemy porc. Most AA units don't. Same with placeholder. That should be easy to fix though, just add porc to their fleeables.

**Comment by mojjj** [comment]
with (old)f-move i meant (current-)attack move. using 'a'-key. corrected the 1st post.

**Comment by GoogleFrog** [comment]
Vandal correctly avoids LLT. Placeholder doesn't avoid LLT. Tasks:
 * Make Placeholder flee static defenses.
 * Check that all mobile AA flees all static defense.

Not addressing the Rogue prediction issue here.

**Comment by GoogleFrog** [comment]
This is a simple tactical AI config task.

**Comment by sprunk** [comment]
I checked and mobile AA correctly avoid LLT.

**Comment by SteelBlueZK** [comment]
The "simple tactical AI config task" isn't as simple as it may seem. It would ideally just be adding a table to "flees" here: https://github.com/ZeroK-RTS/Zero-K/blob/master/LuaRules/Configs/tactical_ai_defs.lua#L1510 . But copying over flea (spiderscout) solution does not work. This is the end of my definitive knowledge. 

This is probably because placeholder (as it should) does not attack turrets https://github.com/ZeroK-RTS/Zero-K/blob/master/units/jumpblackhole.lua#L76 and therefore probably doesn't even see them as far as unitAI is concerned. I tested adding a fake weapon but I don't understand how the engine handles multiple weapons, or how to counter targeting issues that arise from having a fake weapon. I would imagine it requires careful LUS work as well.
